app/content/profile/index.js

```diff
@@ -65,6 +85,147 @@ function Index(props) {
     isLoading: drillInfoIsLoading,
   } = useDrillInfo();
 
+  // ref
+  const bottomSheetModalRef = useRef(null);
+
+  // variables
+  const initialSnapPoints = useMemo(() => [355, 455, 730], []);
+  const expandedSnapPoints = useMemo(() => [460, 570, 860], []); // Adjusted snap points for expanded content
+  const [snapPoints, setSnapPoints] = useState(initialSnapPoints);
+
+  // callbacks
+  const handlePresentModalPress = useCallback(() => {
+    bottomSheetModalRef.current?.present();
+  }, []);
+  const handleSheetChanges = useCallback((index) => {
+    // console.log("handleSheetChanges", index);
+  }, []);
+
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [passwordInputVisible, setPasswordInputVisible] = useState(false);
+
+  const [snackbarVisible, setSnackbarVisible] = useState(false); // State to toggle snackbar visibility
+  const [snackbarMessage, setSnackbarMessage] = useState(""); // State to set snackbar message
+
+  const [dialogVisible, setDialogVisible] = useState(false);
+  const [dialogTitle, setDialogTitle] = useState("");
+  const [dialogMessage, setDialogMessage] = useState("");
+
+  const handleSnackbarDismiss = () => {
+    setSnackbarVisible(false); // Dismiss snackbar
+  };
+
+  useEffect(() => {
+    // Check if userData has been loaded and it contains the name property
+    if (userData && userData.name) {
+      // Update the name state only if it hasn't been set yet
+      if (!name) {
+        setName(userData.name);
+      }
+    }
+  }, [userData, name]); // Watch for changes in userData and name
+
+  useEffect(() => {
```

Comment:
Don't do this, this is really messy. Insert a loading screen at the end of all the hooks so that by the time the code gets here, it's 100% loaded